### PR TITLE
Clean state for uncommitted bundle importer

### PR DESCRIPTION
**Pull request by makyo**
A counter was not reset when a changeset was completed executing, so an attempt to deploy another bundle led to nothing happening (if the counter was > the number of records) or a broken bundle (if the number of records > the counter, but the counter was still not reset).

Addresses https://bugs.launchpad.net/juju-gui/+bug/1454750 and https://bugs.launchpad.net/juju-gui/+bug/1455652

### CODE REVIEW

**Comment by makyo** [comment]
# QA

* `juju set juju-gui juju-gui-source="https://github.com/makyo/juju-gui.git uncommitted-bundles-ecs"`

**Comment by mitechie** [comment]
:+1: code no QA.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/2528/
Test PASSed.

File: app/views/topology/service.js [hunk 1]
```diff
@@ -1191,9 +1191,12 @@ YUI.add('juju-topology-service', function(Y) {
...
       .filter(function(boundingBox) {
-            var annotations = boundingBox.model.get('annotations');
-            return ((!Y.Lang.isNumber(boundingBox.x) &&
-                !(annotations && annotations['gui-x'])));
+            if (boundingBox.model) {
+              var annotations = boundingBox.model.get('annotations');
+              return ((!Y.Lang.isNumber(boundingBox.x) &&
+                  !(annotations && annotations['gui-x'])));
+            }
+            return false;
```
Comment (hatched) [inline app/views/topology/service.js]: What's going on here? Is this the fix for #1454750 ? Could you add a comment to the code as to why this is necessary

File: app/utils/bundle-importer.js [hunk 2]
```diff
@@ -232,6 +232,7 @@ YUI.add('bundle-importer', function(Y) {
           message: 'ChangeSet import complete.',
           level: 'important'
         });
+        this._dryRunIndex = -1;
```
Comment (hatched) [inline app/utils/bundle-importer.js]: Yay! Maybe a quick comment here as to why this is required too plz.

**Comment by hatched** [comment]
:+1: No QA yet

**Comment by hatched** [comment]
It took me three tries but I eventually got a QA OK on this. I don't think that these issues are related to this fix but something we should look at none the less.

Notes:
I followed the QA steps.

The first time I ended up with the following error which is quite odd as I've never seen it before:
screenshot from 2015-05-18 18 35 10

The second time I got this error, which I have seen before, maybe we should put a conditional here. Any idea why this is even possible? Some how the model is undefined when trying to get the annotations.
screenshot from 2015-05-18 18 38 31

Third time worked as advertised.

File: app/views/topology/service.js (hunk 1 above)
Comment (makyo) [inline app/views/topology/service.js]: This was a driveby, meant to put a GH comment on it but spaced it.  Gets rid of one of the three regularly occurring errors with the ECS (the other two you describe below)

**Comment by makyo** [comment]
Thanks for the reviews! :shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/2532/
Test PASSed.
